### ISSUE TITLE
BRs: Clarify how SRVName subjectAltNames should behave

### ISSUE THREAD

**Issue by sleevi** [cabforum/servercert#268]
During the discussion on 2022-04-22 of the draft certificate profile work of the validation subcommittee, one area that was identified was that the requirements of Mozilla's Root Program, with respect to `nameConstraints` and technically constrained sub-CAs, is a superset of the requirements reflected in the Baseline Requirements.

In particular, Mozilla requires restrictions around `rfc822Name` and `SRVName` in order to be considered technically constrained and thus fully out of scope of their program. During the call, the discussion highlighted this was intentional, in order to prevent compatibility or security risks to Mozilla users if/when support for `SRVName`s is introduced for TLS or the adoption of S/MIME BRs.

At issue is the fact that a CA that is technically constrained by the BRs today (via `dNSName`, `iPAddress`, and `directoryName` constraints) would, if clients supported `SRVName`s, be able to issue arbitrary `SRVName`s without being constrained in the same way. Given that, like DNS, `SRVName`s also contain a host name, this is certainly at odds with the goals of technically constrained. By constraining `SRVName`s as well, such sub-CAs can be effectively constrained.

The draft profile work is proposing allowing both `rfc822Name` and `SRVName` name constraints to be added to sub-CAs issued by CAs subject to the Baseline Requirements, but with the requirement that the CA MUST appropriately validate the domain name portion of these fields.

While this explicitly would **not** permit the issuance of these SANs, which are presently forbidden by the BRs and still proposed to be forbidden, as part of the new profiles work, this issue is to track the work that can be done in parallel or subsequently to allow such issuance, as well as to work out what issues clients may have to support such names securely.

**Comment by sleevi** [comment]
New [discussion thread](https://archive.cabforum.org/pipermail/validation/2021-April/001651.html) related to this issue

**Comment by mimi89999** [comment]
Hello,

I uploaded a draft spec for how such certificates could be issued: https://datatracker.ietf.org/doc/draft-lebihan-srv-identifier-validation-extension/

I also started a discussion on the ACME IETF WG mailing list about that: https://mailarchive.ietf.org/arch/msg/acme/wKGYjAirLG8eiuIqHF3cvWQKUY8/

The draft contains a problem statement explaining why such certificates are needed and why alternatives won't work for that use case.
